### PR TITLE
Implement Instant showcase

### PR DESCRIPTION
**Pull request by Cypher1**
Creates a small showcase that uses ArcsInstant and ArcsDuration (types for handling times) in particle code and refinements

### CODE REVIEW

**Review by cromwellian — APPROVED** [review]

**Comment by cromwellian** [comment]
Nice work!

**Review by shans — APPROVED** [review]

File: javatests/arcs/showcase/instant/Calendar.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// LocalDate date = LocalDate.parse("9999-12-31");
+// ArcsInstant instant = date.atStartOfDay(ZoneId.of("Europe/Paris")).toInstant();
+
+typealias Event = AbstractCalendar.Event
+
+@TargetHost(ShowcaseHost::class)
+class Calendar : AbstractCalendar() {
+    override fun onReady() {
+        handles.events.store(
```
Comment (piotrswigon) [inline javatests/arcs/showcase/instant/Calendar.kt:18]: You could use storeAll() and store all 3 events together.

File: javatests/arcs/showcase/instant/Calendar.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            Event(
+                name = "Launch",
+                start = ArcsInstant.ofEpochMilli(819007320000),
```
Comment (piotrswigon) [inline javatests/arcs/showcase/instant/Calendar.kt:21]: A comment for what this value is? Alternatively an `ArcsInstant.parse(textRepresentation)`?

File: javatests/arcs/showcase/instant/EventsToday.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@TargetHost(ShowcaseHost::class)
+class EventsToday : AbstractEventsToday() {
+
+    override fun onReady() {
+        handles.events.fetchAll().forEach {
+            if (
+                (it.start <= ArcsInstant.now().plus(ArcsDuration.ofDays(1))) &&
+                (ArcsInstant.now() <= it.end)
+            ) {
+                handles.agenda.store(
```
Comment (piotrswigon) [inline javatests/arcs/showcase/instant/EventsToday.kt:21]: More Kotlin-y would be:

`handles.agenda.storeAll(handles.events.fetchAll().filter {...}.map {...}`

File: javatests/arcs/showcase/instant/InstantShowcaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        val anHourFromNow = ArcsInstant.now().plus(ArcsDuration.ofHours(1)).toEpochMilli()
+        val twoHoursFromNow = ArcsInstant.now().plus(ArcsDuration.ofHours(2)).toEpochMilli()
+
+        assertThat(teamMeet.start.toEpochMilli() - anHourFromNow).isAtMost(1000)
+        assertThat(teamMeet.end.toEpochMilli() - twoHoursFromNow).isAtMost(1000)
+
+        delay(1000) // This where the arc.idle should be called
+
+        val eventsParticle: EventsToday = env.getParticle<EventsToday>(arc)
+        val todaysEvents = eventsParticle.handles.events.dispatchFetchAll()
```
Comment (piotrswigon) [inline javatests/arcs/showcase/instant/InstantShowcaseTest.kt:72]: Do you want to assert that there's only one even in today's agenda?

Comment (Cypher1) [reply javatests/arcs/showcase/instant/InstantShowcaseTest.kt]: Yes! Thanks, this caught a mistake where I was checking the wrong handle.

File: javatests/arcs/showcase/instant/Calendar.kt (hunk 1 above)
Comment (Cypher1) [reply javatests/arcs/showcase/instant/Calendar.kt]: I'd have to construct a collection to do that and there are no other showcases doing that at the moment.

If possible I'll just demo Instant, not introduce other features.

File: javatests/arcs/showcase/instant/Calendar.kt (hunk 2 above)
Comment (Cypher1) [reply javatests/arcs/showcase/instant/Calendar.kt]: Added comments with the dates. ArcsInstant doesn't currently have a .parse. I'll add a jvm impl for it next then start on Duration.

**Pull request closed without merging**